pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2518,6 +2549,35 @@ func handlePhysicalIOAdapterListImpl(ctxArg interface{}, key string,
 			// We assume AddOrUpdateIoBundle will preserve any
 			// existing IsPort/IsPCIBack/UsedByUUID
 			aa.AddOrUpdateIoBundle(log, *ib)
+
+			// Adding PF before VF to have correct boot order
+			if ib.Type == types.IoNetEthPF {
+				vfCnt, err := vfCount(phyAdapter)
+				if err != nil {
+					log.Functionf("Failed to get vfCount for phyAdapter %s", err)
+				}
+				if vfCnt > 0 {
+					err, vfs := createAndGetVF(ib.Ifname, vfCnt)
+					if vfs == nil {
+						log.Fatal("Failed to create VF for iface", ib.Ifname)
```

Comment:
Can't this fail if I pass in vfCnt=maxInt? If so it shouldn't be a log fatal but instead record the failure so it can be returned to the controller. And it would make sense to include vfCnt in the error/log message.

Reply:
Wouldn't err be returned in this case?

Reply:
It would have been, fixed it